src/service/video-manager-impl.js

```diff
@@ -330,11 +430,30 @@ class VideoEntry {
    * @private
    */
   loadedVideoVisibilityChanged_() {
-    if (this.hasAutoplay_) {
+    if (this.hasAutoplay) {
       this.autoplayLoadedVideoVisibilityChanged_();
     }
   }
 
+  /* Docking Behaviour */
+
+  /**
+   * Called when a dockable video is built.
+   * @private
+   */
+  dockableVideoBuilt_() {
+    this.video.element.classList.add('i-amphtml-dockable-video');
+
+    const vidRect = this.video.element./*OK*/getBoundingClientRect();
+
+    this.initialWidth_ = vidRect.width;
```

Comment:
cache the whole 'rect' instead of individual values? (to reduce number of instance variable we have, mostly for readibility )